internal/transport/http2_client.go

```diff
@@ -468,6 +472,7 @@ func newHTTP2Client(connectCtx, ctx context.Context, addr resolver.Address, opts
 			t.conn.Close()
 		}
 		close(t.writerDone)
+		t.writerMu.Unlock()
```

Comment:
Wouldn't that cause deadlock? Let's say we acquire lock to initialise it and we release it. Now it acquires lock for [this](https://github.com/grpc/grpc-go/pull/7015/files#diff-5334018444bcb34d15bcb3c8a3050cdf9c8a40fe41305c3a602ea4bfeb0cd178R1013) and since the lock is already acquired, it will never execute [this](https://github.com/grpc/grpc-go/pull/7015/files#diff-5334018444bcb34d15bcb3c8a3050cdf9c8a40fe41305c3a602ea4bfeb0cd178R474)